hir-type/src/lib.rs

```diff
@@ -65,6 +65,8 @@ pub enum Type {
     Struct(StructType),
     /// A vector of fixed size
     Array(Box<Type>, usize),
+    /// A tuple of values of the given types
```

Comment:
Do we actually need a `Tuple` type? A tuple is basically just an anonymous struct, there is no meaningful difference between them. I'd rather avoid introducing additional type-level abstractions if we can avoid it, since it requires handling that type in various places throughout the compiler, and we don't really want to do that unless it is actually semantically significant.

If there is a specific reason why `StructType` is insufficient, let me know, but for now I'm going to leave it out of the rebased changes and use structs instead.

Reply:
Probably, not. I cannot find any reason why it cannot be represented as a struct.